ee/clickhouse/queries/trends/clickhouse_trends.py

```diff
@@ -52,6 +53,45 @@ def _run_query(self, filter: Filter, entity: Entity, team_id: int) -> List[Dict[
             serialized_data = self._handle_cumulative(serialized_data)
         return serialized_data
 
+    def _run_query_for_threading(self, result: List, index: int, sql, params):
+        result[index] = sync_execute(sql, params)
+
+    def _run_parallel(self, filter: Filter, team: Team) -> List[Dict[str, Any]]:
+        result: List[Union[None, List[Dict[str, Any]]]] = [None] * len(filter.entities)
+        jobs = []
+
+        for entity in filter.entities:
+            sql, params, _ = self._get_sql_for_entity(filter, entity, team.pk)
```

Comment:
Q: Why not generate the sql/params in the thread as well? We could leverage `_run_query` (indirectly) then.